### ISSUE TITLE
Disney+ not muting and/or not unmuting

### ISSUE THREAD

**Issue by duncan1a** [FrostCo/AdvancedProfanityFilter#540]
#### :bug: Description
I'm using Edge and disney+ from Ireland. If APF is set to mute the tab, then it doesn't mute profanity. If It's set to any of the other options (mute video, mute volume) then it does mute but doesn't unmute again

#### :twisted_rightwards_arrows: Steps To Reproduce
1. Go to Advanced profanity filter Options
2. Click on Audio
3. Set Mute Method to Browser Tab - play Disney+ video - does not mute
4. Set Mute method to Mute Video or Mute Volume - refresh Disney+ video - mutes but doesn't unmute again

Note, I have also set Captions/Subtitles to Hide All

#### :heavy_check_mark: Expected behavior
Expect to mute and profanity and then unmute video

#### :computer: Details
 - **Device:** M3 Mackbook Pro
 - **OS:** Sonoma 14.2.1
 - **Browser:** Edge
 - **Browser Version:** 20.0.2210.144
 - **Affected site(s):** www.disneyplus.com

#### :pencil: Additional context
Audio works as expected on other sites

**Comment by richardfrost** [comment]
Hi @duncan1a, I'm sorry muting isn't working on Disney+ for you. I will see if I can replicate the issue in an effort to hopefully fix it. I'll let you know how it goes. One follow-up question (thanks for providing all that info by the way, it was very helpful!): Has it worked on Disney+ for you before?

**Comment by duncan1a** [comment]
We've just signed up for Disney, so this is the first time we've tried. It works perfectly for paramount+ and Youtube.

Thank you for this project, by the way. It really does make a big difference

**Comment by richardfrost** [comment]
Thanks for the additional details @duncan1a. I'm wondering if Disney+ Ireland is different than US/UK. South Africa was, but I was able to work with someone there to get it working. If you'd be willing to help me out I can definitely give it a try for Ireland too. If you're interested, please send an email to support @ frostco.dev (without the spaces) and  we can discuss it more. In the meantime, I'll continue to see if I can figure anything out on my end.

**Comment by kwiffen** [comment]
I’ve been having the same issues with Disney+ here in Canada too. Just curious if you have any suggestions? Thanks for your help! 🙏🏻

**Comment by richardfrost** [comment]
Hi @kwiffen, thanks for reporting on this issue. I just tried a test connected to a VPN in Canada and it worked, but that might be because I'm using a U.S. account. Has it ever worked for you before on Disney+? Any other information you have could be helpful. If you have a specific video you are trying on I can test that one too, so hopefully we can find what is going on and fix it.

**Comment by kwiffen** [comment]
Thank you! Just tried it, and it mutes/removes the subtitles, but it won’t unmute afterwards. Is there something I can do to fix this in my end? Thanks again!Kristine Sent from my iPhoneOn Feb 11, 2024, at 2:53 PM, Richard Frost ***@***.***> wrote:﻿
Hi @kwiffen, thanks for reporting on this issue. I just tried a test connected to a VPN in Canada and it worked, but that might be because I'm using a U.S. account. Has it ever worked for you before on Disney+? Any other information you have could be helpful. If you have a specific video you are trying on I can test that one too, so hopefully we can find what is going on and fix it.

**Comment by richardfrost** [comment]
Thanks for getting back to me @kwiffen. Is it muting when it should? Or as soon as it sees any captions/subtitles? Could you share what video (and the approximate timestamp) for when its muting?

One other thing you might try, a few weeks back I updated the config for Disney+ and it possible that could have been causing some issues. Could you try using this custom config and see if it is any different?

```json
{
  "www.disneyplus.com": [
    {
      "className": "dss-subtitle-renderer-wrapper",
      "mode": "element",
      "subtitleSelector": "div.dss-subtitle-renderer-cue-window span.dss-subtitle-renderer-line",
      "tagName": "DIV"
    }
  ]
}
```

To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration.

**Comment by kwiffen** [comment]
It worked! Thank you so much! Love the program. We watch everything through the filter, and it has allowed us to watch things we would otherwise not have been able to watch. Some movies and shows have a bit of swearing but nothing else offensive. So unnecessary! Thank you again!KristineSent from my iPhoneOn Feb 11, 2024, at 5:30 PM, Richard Frost ***@***.***> wrote:﻿
Thanks for getting back to me @kwiffen. Is it muting when it should? Or as soon as it sees any captions/subtitles? Could you share what video (and the approximate timestamp) for when its muting?
One other thing you might try, a few weeks back I updated the config for Disney+ and it possible that could have been causing some issues. Could you try using this custom config and see if it is any different?
{
  "www.disneyplus.com": [
    {
      "className": "dss-subtitle-renderer-wrapper",
      "mode": "element",
      "subtitleSelector": "div.dss-subtitle-renderer-cue-window span.dss-subtitle-renderer-line",
      "tagName": "DIV"
    }
  ]
}
To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration.

**Comment by TruSteel** [comment]
Mute would work at a curse word but would stayed muted till I reset filter. Next curse word it would do it again. So I pasted this code in for Disney plus to get mute to work. This config works great.
{
  "www.disneyplus.com": [
    {
      "className": "dss-subtitle-renderer-wrapper",
      "mode": "element",
      "subtitleSelector": "div.dss-subtitle-renderer-cue-window span.dss-subtitle-renderer-line",
      "tagName": "DIV"
    }
  ]
}

**Comment by anaquialex** [comment]
for me it worked, on audio i switched from browser tab to video mute. the only thing that wont block in directly on the hulu website. but i does block if i watch hulu on disney plus. thank you for this software, it makes it watching movies and tv shows enjoyable.

**Comment by duncan1a** [comment]
That worked for me too

**Comment by richardfrost** [comment]
Disney+ has updated their site again, and the latest config can be found in issue #595. This should now be fixed with version [4.0.0](https://github.com/FrostCo/AdvancedProfanityFilter/releases/tag/4.0.0). If anyone is still having issues, please respond in #595.
